myia/abstract/infer.py

```diff
@@ -315,12 +317,15 @@ def get_inferrer_for(self, mg: MetaGraphFunction):
                 f'{fn.values[VALUE]} {fn.values[DATA] or ""}.'
             )
 
-        elif not isinstance(fn, AbstractFunction):
+        elif isinstance(fn, AbstractClassBase):
             g = ref.node.graph
```

Comment:
Is that to be able to call classes?

Reply:
Calling dataclass instances if they define `__call__`, yes.